hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -164,6 +163,34 @@ class ExpressionPayload(@transient record: GenericRecord,
     }
   }
 
+  private def doRecordMerge(incomingRecord: GenericRecord,
+                            existingRecord: IndexedRecord,
+                            schema: Schema,
+                            properties: Properties): HOption[IndexedRecord] = {
+    val tablePayloadClass = properties.getProperty(PAYLOAD_ORIGINAL_AVRO_PAYLOAD)
+    if (tablePayloadClass.equals(classOf[OverwriteWithLatestAvroPayload].getName)) {
+      HOption.of(incomingRecord)
+    } else if (tablePayloadClass.equals(classOf[DefaultHoodieRecordPayload].getName)) {
+      if (needUpdatingPersistedRecord(existingRecord, incomingRecord, properties)) {
```

Comment:
can you add java docs for needUpdatingPersistedRecord while you are at it.

Reply:
there is some more logic in combineAndGetUpdateValue of DefaultHoodieRecordPayload which we are bypassing here. it might be better if we call into the DefaultHoodieRecordPayload.combineAndGetUpdateValue only.